user/tests/wiring/no_fixture/acm.cpp

```diff
@@ -0,0 +1,155 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+#include "application.h"
+#include "unit-test/unit-test.h"
+
+const uint32_t WIFI_CONNECT_TIMEOUT_MS = 5 * 60 * 1000;
+const uint32_t DISCONNECT_TIMEOUT_MS = 5 * 60 * 1000;
+
+test(ACM_01_PREPARE) {
+    Particle.disconnect();
+
+#if HAL_PLATFORM_WIFI
+    WiFi.disconnect();
+#endif // HAL_PLATFORM_WIFI
+#if HAL_PLATFORM_CELLULAR
+    Cellular.disconnect();
+#endif // HAL_PLATFORM_CELLULAR
+#if HAL_PLATFORM_ETHERNET
+    Ethernet.disconnect();
+#endif // HAL_PLATFORM_ETHERNET
+
+    assertEqual(Network, Particle.connectionInterface());
+}
+
+test(ACM_02_particle_connect_uses_preferred_network) {
+    auto expectedNetwork = Network;
+
+#if HAL_PLATFORM_WIFI && !HAL_PLATFORM_WIFI_SCAN_ONLY
+    expectedNetwork = WiFi;
+    WiFi.prefer();
+    WiFi.connect();
+    assertTrue(waitFor(WiFi.ready, WIFI_CONNECT_TIMEOUT_MS));
+#endif // HAL_PLATFORM_WIFI && !HAL_PLATFORM_WIFI_SCAN_ONLY
+
+#if HAL_PLATFORM_CELLULAR
+    expectedNetwork = Cellular;
+    Cellular.prefer();
+    Cellular.connect();
+    assertTrue(waitFor(Cellular.ready, HAL_PLATFORM_CELLULAR_CONN_TIMEOUT));
+#endif // HAL_PLATFORM_CELLULAR
+
+    Particle.connect();
+    assertTrue(waitFor(Particle.connected, HAL_PLATFORM_MAX_CLOUD_CONNECT_TIME));
+    assertEqual(Particle.connectionInterface(), expectedNetwork);
+}
+
+
+#if PLATFORM_ID == PLATFORM_MSOM
+test(ACM_03_prefer_one_network_only) {
+	WiFi.prefer();
+	assertFalse(Cellular.isPreferred());
+	Cellular.prefer();
+	assertFalse(WiFi.isPreferred());
+}
+
+test(ACM_04_prefer_moves_cloud_connection_when_set) {
+    // Default preference
+	Network.prefer();
+	
+    // Connect all interfaces
+    Network.connect();
+    Particle.connect();
+    assertTrue(waitFor(Particle.connected, HAL_PLATFORM_MAX_CLOUD_CONNECT_TIME));
+
+    auto connectedNetwork = Particle.connectionInterface();
+
+    // Prefer whatever network we did not connect on in order to force the cloud connection to switch
+    if (connectedNetwork == Cellular) {
+	    assertTrue(waitFor(WiFi.ready, WIFI_CONNECT_TIMEOUT_MS));
+    	WiFi.prefer();
+    } else {
+    	assertTrue(waitFor(Cellular.ready, HAL_PLATFORM_CELLULAR_CONN_TIMEOUT));
+    	Cellular.prefer();
+    }
+
+    // Verify that we disconnect and reconnect on the newly preferred network
+    assertTrue(waitForNot(Particle.connected, DISCONNECT_TIMEOUT_MS));
+    assertTrue(waitFor(Particle.connected, HAL_PLATFORM_MAX_CLOUD_CONNECT_TIME));
+    assertNotEqual(connectedNetwork, Particle.connectionInterface());
+}
+
+test(ACM_05_cloud_connection_moves_to_preferred_network_when_available) {
+    // Disconnect WiFi
+    WiFi.disconnect();
+    Particle.disconnect();
+    assertTrue(waitForNot(Particle.connected, DISCONNECT_TIMEOUT_MS));
+    assertTrue(waitForNot(WiFi.ready, DISCONNECT_TIMEOUT_MS));
```

Comment:
Shouldn't we Particle.disconnect, wait for that, then WiFi.disconnect and wait for that?